loopy/transform/iname.py

```diff
@@ -2404,6 +2405,16 @@ def does_insn_involve_iname(kernel, insn, *args):
             smap.map_kernel(kernel, within=does_insn_involve_iname,
                             map_tvs=False, map_args=False))
 
+    # replace instances where the old inames appear as a param
+    new_domains = []
+    for dom in kernel.domains:
+        for old_iname in orig_old_inames:
+            d = dom.get_var_dict()
+            if old_iname in d and new_iname not in d:
+                var_type, var_num = d[old_iname]
+                dom = dom.set_dim_name(var_type, var_num, new_iname)
+        new_domains.append(dom)
```

Comment:
As it stands, this is not quite correct. The old iname may survive, and it may remain in use (if `within` is given). In that case, I suggest we raise an (not-implemented) error if the iname occurs as a parameter to another domain.